### PR TITLE
fix(rust): node duplication error

### LINKED ISSUE #4492: Creating a node twice results in an unclear error
### Observed behavior

A not very explicit error when trying to create a node twice:
```
IO error. Caused by: File exists (os error 17)
```

### Steps to reproduce

I create a node with a specific name
```
❯ ockam node create n1                                                                                                                                                                                                                                                                                                                                                                                                        ockam default

Node:
  Name: n1
  Status: UP
  Route To Node:
    Short: /node/n1
    Verbose: /dnsaddr/localhost/tcp/65125
....
```

And try to create it again
```
❯ ockam node create n1                                                                                                                                                                                                                                                                                                                                                                                                        ockam default
Version 0.82.0, hash: fbdf9e37fe73195e8f8f9db7d908f38fbad2d121
IO error. Caused by: File exists (os error 17)
```

### Expected behavior

I would be nice to query the list of existing nodes and report that a node already exists with the same name instead of failing to re-created it.

### Ockam Version

```
0.82.0  Compiled from (git hash): fbdf9e37fe73195e8f8f9db7d908f38fbad2d121
```

### PR DESCRIPTION
**Pull request by MehSalhi**
Added code to check if a node already exists with the same name in the default node folder when trying to create a new node with ockam node create <name>

## Current behavior

When creating a new node, if another node already exists with that name, the current error is : 

`IO error. Caused by: File exists (os error 17)`

This error is unclear.

## Proposed changes

Should fix #4492 

Check if a node with that name exists before trying to create it by checking in the node directory. Return a more appropriate error if name already exists.

Error used : CliStateError::AlreadyExists

Output with this fix: 

`
already exists: node with name node_name already exists
`

## Checks

- [X] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [X] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [X] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [X] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/cli_state.rs [hunk 1]
```diff
@@ -689,6 +689,12 @@ impl NodesState {
 
     pub fn create(&self, name: &str, mut config: NodeConfig) -> Result<NodeState> {
         config.name = name.to_string();
+
+        // check if node name already exists
+        if self.dir.join(name).exists() {
+            return Err(CliStateError::AlreadyExists(format!("node with name `{}` already exists", name)));
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_api/src/cli_state.rs]: To keep the error consistent, let's change the text to:

```return Err(CliStateError::AlreadyExists(format!("node `{name}`")));```

**Review by adrianbenavides — CHANGES_REQUESTED** [review]
Hey @MehSalhi , thanks for submitting your first PR to Ockam!

I left a comment, and also make sure to run `cargo fmt` to fix the formatting issues.

File: implementations/rust/ockam/ockam_api/src/cli_state.rs (hunk 1 above)
Comment (MehSalhi) [reply implementations/rust/ockam/ockam_api/src/cli_state.rs]: Thank you for the feedback. I made the changes.

**Review by adrianbenavides — APPROVED** [review]
Perfect, thank you so much @MehSalhi, LGTM! 🥳

**Pull request merged**